### PR TITLE
Top level metrics and params

### PR DESCRIPTION
**Pull request by dberenbaum**
Quick fix to add in some basic reference info for top-level metrics and params.

@skshetry Could you please review and take this one over so we can get it merged today 🙏 ?

### CODE REVIEW

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 1]
```diff
@@ -12,14 +13,113 @@ modify, write, or generate them by your own means.
...
+params:
+  - params.yaml
+```
+
+Parameters are key/value pairs saved in structured files. Unlike stage-level
+[parameter dependencies](#parameters), which are granular, top-level parameters
+are defined at the file level and include all parameters in the file. See
+`dvc params` for more information and how to compare between experiments.
+
+## Plots
```
Comment (dberenbaum) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:45]: Moved this up to have the top-level sections in alphabetical order at the top and the more granular stage-specific info towards the bottom. Plots is untouched here besides being moved up.

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 2]
```diff
@@ -12,14 +13,113 @@ modify, write, or generate them by your own means.
...
+
+The list of `params` contains one or more paths to <abbr>parameters</abbr>
+files. Here's an example:
+
+```yaml
+params:
+  - params.yaml
+```
+
+Parameters are key/value pairs saved in structured files. Unlike stage-level
```
Comment (daavoo) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:40]: Wait, top-level params can't be granular?

**Review by daavoo — APPROVED** [review]

**Comment by daavoo** [comment]
Merging the current state. Let's address any comment in follow-up

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 3]
```diff
@@ -12,14 +13,113 @@ modify, write, or generate them by your own means.
...
+Here's an example:
+
+```yaml
+metrics:
+  - metrics.json
+```
+
+Metrics are key/value pairs saved in structured files that map a metric name to
+a numeric value. See `dvc metrics` for more information and how to compare among
+experiments.
```
Comment (skshetry) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:28]: I don't know if we should be pedantic, but metrics don't necessarily have to be key/value pairs. For example, `metrics.json` could just be:
```json
3.14
```

```console
$ dvc metrics show
Path                                                                  
metrics.json  3.14
```

They could be list too, but that seems unsupported at the moment (there's no reason not to support them though)
```yaml
- avg_prec: 0.9249974999612706
  roc_auc: 0.9460213440787918
- avg_prec: 0.9249974999612706
  roc_auc: 0.9460213440787918 
```

**Comment by skshetry** [comment]
Looks good, thanks @dberenbaum. I don't see anything that we need to add here.

File: content/docs/user-guide/project-structure/dvcyaml-files.md (hunk 3 above)
Comment (dberenbaum) [reply content/docs/user-guide/project-structure/dvcyaml-files.md:28]: 🤔 That's new to me. I'm struggling to imagine how it's useful without a key. How would you show the list example in `metrics show`?

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 4]
```diff
@@ -1,9 +1,10 @@
...
 
-You can construct machine learning pipelines by defining individual
-[stages](/doc/command-reference/run) in one or more `dvc.yaml` files. Stages
-constitute a pipeline when they connect with each other (forming a [dependency
-graph], see `dvc dag`).
+You can configure machine learning projects in one or more `dvc.yaml` files. The
+list of [`stages`](#stages) is typically the most important part of a `dvc.yaml`
+file, though the file can also be used to configure [`metrics`](#metrics),
+[`params`](#params), and [`plots`](#plots), either as part of a stage definition
+or on their own.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:7]: configure -> define?
part of a stage ~~definition~~ ?

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 5]
```diff
@@ -12,14 +13,113 @@ modify, write, or generate them by your own means.
...
+```yaml
+metrics:
+  - metrics.json
+```
+
+Metrics are key/value pairs saved in structured files that map a metric name to
+a numeric value. See `dvc metrics` for more information and how to compare among
+experiments.
+
+## Params
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:30]: Parameters? (change **Stage** -> **Parameters** to **Parameter dependencies**)

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 6]
```diff
@@ -58,8 +158,7 @@ them).
 
 <admon type="tip">
 
-Output files may be viable data sources for
-[top-level plots](#top-level-plot-definitions).
+Output files may be viable data sources for [top-level plots](#plots).
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:161]: ~~top-level~~ ?

**Review by jorgeorpinel — COMMENTED** [review]
A few minor wording clarification ideas 👇🏼

File: content/docs/user-guide/project-structure/dvcyaml-files.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/project-structure/dvcyaml-files.md:45]: Pretty smart edit 👏🏼

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 7]
```diff
@@ -12,14 +13,113 @@ modify, write, or generate them by your own means.
...
+```
+
+Metrics are key/value pairs saved in structured files that map a metric name to
+a numeric value. See `dvc metrics` for more information and how to compare among
+experiments.
+
+## Params
+
+The list of `params` contains one or more paths to <abbr>parameters</abbr>
+files. Here's an example:
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:33]: The `parameters` tooltip refers to param deps specifically (see https://github.com/iterative/dvc.org/issues/4202#issuecomment-1363466618).

**Comment by jorgeorpinel** [comment]
> let go some of these if they are low impact

@dberenbaum the only possibly impactful issue I see is #4202 but there's an issue for that so we can address it there. Doesn't look urgent to me (unless users start reporting confusion). Thanks

**Pull request merged**